From 2e45806ad37f8ff7b0a6ac0d0e9ec323ae5617a8 Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Sat, 8 Jul 2023 12:40:36 -0500
Subject: drivers/iommu: Add GRR device ID

Until updates to the Intel iommu driver that include support the GRR
device ID have been upstreamed, add the ID here.

Signed-off-by: John Jacques <john.jacques@intel.com>
---
 drivers/iommu/intel/iommu.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/iommu/intel/iommu.c b/drivers/iommu/intel/iommu.c
index c35751a15179..b897e481502f 100644
--- a/drivers/iommu/intel/iommu.c
+++ b/drivers/iommu/intel/iommu.c
@@ -63,6 +63,7 @@
 			    (pdev)->device == 0x465d ||		\
 			    (pdev)->device == 0x1919))
 #define IS_AZALIA(pdev) ((pdev)->vendor == 0x8086 && (pdev)->device == 0x3a3e)
+#define IS_GRR(pdev) ((pdev)->vendor == 0x8086 && (pdev)->device == 0x0dbd)
 
 #define IOAPIC_RANGE_START	(0xfee00000)
 #define IOAPIC_RANGE_END	(0xfeefffff)
@@ -3029,6 +3030,9 @@ static int device_def_domain_type(struct device *dev)
 
 		if ((iommu_identity_mapping & IDENTMAP_IPU) && IS_INTEL_IPU(pdev))
 			return IOMMU_DOMAIN_IDENTITY;
+
+		if (IS_GRR(pdev))
+			return IOMMU_DOMAIN_IDENTITY;
 	}
 
 	return 0;
-- 
2.25.1

